packages/liveblocks-core/src/auth-manager.ts

```diff
@@ -35,7 +35,8 @@ export type AuthenticationOptions = {
 );
 
 export function createAuthManager(
-  authOptions: AuthenticationOptions
+  authOptions: AuthenticationOptions,
+  onAuthenticate?: (token: AuthToken) => void
```

Comment:
I added a simple `onAuthenticate` callback to the `createAuthManager` function. We update the `currentUserIdStore` on each call to this callback.

```ts
const currentUserIdStore = createStore<string | undefined>(undefined);

const authManager = createAuthManager(options, (token) => {
  const userId = token.k === TokenKind.SECRET_LEGACY ? token.id : token.uid;
  currentUserIdStore.set(() => userId);
});
```